arch-chroot.in

```diff
@@ -30,53 +30,6 @@ itself to make it one, i.e. 'mount --bind chroot_dir chroot_dir'.
 EOF
 }
 
-resolve_link() {
-  local target=$1
-  local root=$2
-
-  # If a root was given, make sure it ends in a slash.
-  [[ -n $root && $root != */ ]] && root=$root/
-
-  while [[ -L $target ]]; do
-    target=$(readlink -m "$target")
-    # If a root was given, make sure the target is under it.
-    # Make sure to strip any leading slash from target first.
-    [[ -n $root && $target != $root* ]] && target=$root${target#/}
-  done
-
-  printf %s "$target"
-}
-
-chroot_add_resolv_conf() {
-  local chrootdir=$1
-  local src=$(resolve_link /etc/resolv.conf)
-  local dest=$(resolve_link "$chrootdir/etc/resolv.conf" "$chrootdir")
```

Comment:
The commit moves some code around while also changes the functionality. These two should be separate commits.

The commit message mentions "chroot_add_resolve_link is cleaned up to be simpler" where chroot_add_resolve_link does not exist in either new or old code. Ideally the message will mention what or why the new approach is simpler.